packager/media/formats/webm/multi_segment_segmenter.cc

```diff
@@ -77,15 +88,16 @@ Status MultiSegmentSegmenter::DoFinalize() {
 Status MultiSegmentSegmenter::NewSegment(uint64_t start_timestamp,
                                          bool is_subsegment) {
   if (!is_subsegment) {
-    // Create a new file for the new segment.
-    std::string segment_name =
+    temp_file_name_ =
```

Comment:
It should be either a memory file (prefix with "memory://") or an actual temporary file:

```c++
  if (!TempFilePath(options().temp_dir, &temp_file_name_))
    return Status(error::FILE_FAILURE, "Unable to create temporary file.");
```

This is because `options().segment_template` could be a remote file, e.g. HttpFile. We don't want to create a temporary file there.

Reply:
Sorry about that. I have made the change.

Reply:
@kqyang Do you want to change temp_file_name_ to memory_file_name_?

Reply:
No. It is fine with me. Thanks.